src/Microsoft.ML.Recommend/IMIValueMapper.cs

```diff
@@ -0,0 +1,93 @@
+﻿//------------------------------------------------------------------------------
+// <copyright company="Microsoft Corporation">
+//     Copyright (c) Microsoft Corporation. All rights reserved.
+// </copyright>
+//------------------------------------------------------------------------------
+
+using System;
+using Microsoft.ML.Runtime.Data;
+
+namespace Microsoft.ML.Runtime.Recommend
+{
+    /// <summary>
+    /// Interface for mapping a two input value (of the two indicated ColumnTypes) to
+    /// an output value (of an indicated ColumnType). This interface is commonly implemented
+    /// by predictors. Note that the input and output ColumnTypes determine the proper
+    /// type arguments for GetMapper, but typically contain additional information like
+    /// vector lengths.
+    /// </summary>
+    public interface IMIValueMapper
```

Comment:
>IMIValueMapper [](start = 21, length = 14)

I think it is premature to make this interface.

Reply:
That wasn't done by me, but I can try to clean them up after IEstimator and ITransformer are finished. Could you add more details about your idea? Do you only want to remove IMIValueMapper?

Reply:
I don't feel it's a part of this PR. Please see my comment below. Thanks.

---
In reply to: [225272314](https://github.com/dotnet/machinelearning/pull/1263#discussion_r225272314) [](ancestors = 225272314)